core/shared/platform/windows/platform_internal.h

```diff
@@ -146,8 +146,19 @@ typedef enum windows_access_mode {
     windows_access_mode_write = 1 << 1
 } windows_access_mode;
 
+// These enum values are defined to be the same as the corresponding WASI
+// fdflags so they can be used interchangeably.
+typedef enum windows_fdflags {
```

Comment:
Why do we even define the enum instead of using `__wasi_fdflags_t`? IIRC we already moved it to `platform_wasi.h`

Reply:
I'm not fully convinced that `platform_wasi.h` should be dependent on `platform_internal.h`. Or if it is, we should probably have another file with just with WASI definitions. 
We could also move the os_file* / os_clock* APIs to `platform_api_extension.h` file.

Reply:
Yes, that's true. We could separate out the WASI types completely from `platform_wasi.h` into a separate header so that `platform_wasi.h` doesn't depend on `platform_internal.h`.This is what I originally suggested - https://github.com/bytecodealliance/wasm-micro-runtime/pull/2585#discussion_r1363623529 but I think we subsequently agreed to keep the WASI types and os_file_*/os_clock_* declarations in the same header. 

I'm happy to change it so we have a header with just WASI types. - I'm just not sure of the benefit (apart from reducing this small code duplication). As mentioned, in general, there is no reason why `platform_internal.h` should use or depend on WASI types.

@wenyongh perhaps you have some further thoughts on this?

Reply:
Agree to separate out the WASI types to a file, how about `platform_wasi_types.h`? And should we rename platform_wasi.h?

Reply:
Ok sure - how about we move the filesystem/clock function declarations to `platform_api_extension.h` as @mkolny suggested and we rename `platform_wasi.h` to `platform_wasi_types.h`? 

Also, I'd rather make these changes in a follow-up PR to avoid squishing them in here as it seems it's a separate issue to enabling the WASI tests in Windows. Then I can remove `windows_fdflags` in the same follow-up PR.

Reply:
> Also, I'd rather make these changes in a follow-up PR to avoid squishing them in here as it seems it's a separate issue to enabling the WASI tests in Windows. Then I can remove windows_fdflags in the same follow-up PR.

That sounds reasonable to me.

Reply:
+1

Reply:
thanks for approving/merging, have opened https://github.com/bytecodealliance/wasm-micro-runtime/pull/2724 for the WASI type header changes